File: Algorithm_70_Climbing Stairs.cpp
#include <iostream>
#include <vector>
using namespace std;
int climbStairs(int n)
{
	if (n==1) return 1;
	else if (n==2) return 2;
	int t1=1;
	int t2=2;
	int tmp;
	for (int i=0;i<n-3;i++)
	{
		tmp=t1;
		t1=t2;
		t2+=tmp;
	}
	return t1+t2;
}
int main()
{
	cout<<climbStairs(3);
	return 0;
}

File: Algorithm_70_Climbing Stairs.java
// 思路：直接动态规划，dp[i] = dp[i - 1] + dp[i - 2]
// 此处优化为O(1)空间复杂度

class Solution {
    public int climbStairs(int n) {
        int last_1 = 1, last_2 = 1, cur = 1;
        for (int i = 2; i <= n; i++) {
            cur = last_1 + last_2;
            last_2 = last_1;
            last_1 = cur;
        }
        return cur;
    }
}